internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -83,48 +80,47 @@ func New(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOpti
 	var err error
 	r.proxyURL, err = mapAddress(target.Endpoint())
 	if err != nil {
-		return nil, fmt.Errorf("failed to determine proxy URL for target : %v", err)
+		return nil, fmt.Errorf("failed to determine proxy URL for %v  target endpoint: %v", target.Endpoint(), err)
 	}
 
 	if r.proxyURL == nil {
-		logger.Info("No proxy URL detected")
+		if logger.V(2) {
+			logger.Info("No proxy URL detected")
+		}
 		return targetResolverBuilder.Build(target, cc, opts)
 	}
 
-	// When the scheme is 'dns', resolution should be handled by the proxy, not
-	// the client. Therefore, we bypass the target resolver and store the
-	// unresolved target address.
+	// When the scheme is 'dns' and target resolution on client is not enabled
+	// (can be enabled by the WithTargetResolutionEnabled() dialOption), resolution should be handled by thevproxy, not the client. Therefore, we bypass the target resolver and
```

Comment:
no need to mention can be enabled part i think

Reply:
Done.